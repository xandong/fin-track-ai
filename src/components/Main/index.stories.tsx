import { Meta, StoryObj } from "@storybook/react"
import Main from "."

const meta: Meta = {
  title: "Main",
  component: Main,
  parameters: {
    layout: "fullscreen"
  }
} as Meta

export default meta

type Story = StoryObj<typeof Main>

export const Default: Story = {}
