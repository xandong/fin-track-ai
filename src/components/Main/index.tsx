import { UserButton } from "@clerk/nextjs"

const foo = "bla"

const Main = () => {
  return (
    <main className="flex h-[100vh] items-center justify-center bg-[#0A0A0A]">
      <h1 className="text-5xl font-bold text-white">Main Component</h1>

      <div className="text-white">
        <UserButton showName />
      </div>
    </main>
  )
}

export default Main
