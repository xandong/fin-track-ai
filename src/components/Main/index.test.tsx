import { render, screen } from "@testing-library/react"

import Main from "."

describe("<Main />", () => {
  it("should render the element", () => {
    // renderiza o component
    const { container } = render(<Main />)

    // busca o elemento e verifica se da match
    expect(
      screen.getByRole("heading", {
        name: /main component/i
      })
    ).toBeInTheDocument()

    // gera snapshot
    expect(container.firstChild).toMatchSnapshot()
  })
})
