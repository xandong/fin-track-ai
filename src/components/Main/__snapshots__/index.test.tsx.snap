// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the element 1`] = `
<main
  class="h-[100vh] bg-[#0A0A0A] flex justify-center items-center"
>
  <h1
    class="text-white text-5xl font-bold"
  >
    Main Component
  </h1>
</main>
`;
